lib/improver/wind_gust_diagnostic.py

```diff
@@ -201,18 +202,16 @@ def process(self, cube_gust, cube_ws):
                    ' {0:s} {1:s}.'.format(perc_coord_gust.name(),
                                           perc_coord_ws.name()))
             raise ValueError(msg)
-        # Add metadata to both cubes
+
+        # Add metadata to gust cube
         req_cube_gust = self.add_metadata(req_cube_gust)
-        req_cube_ws = self.add_metadata(req_cube_ws)
-        # Merge cubes
-        merged_cube = merge_cubes(iris.cube.CubeList([req_cube_gust,
-                                                      req_cube_ws]))
-        # Calculate wind-gust diagnostic
-        cube_max = merged_cube.collapsed(perc_coord_gust.name(),
-                                         iris.analysis.MAX)
+
+        # Calculate wind-gust diagnostic using CubeCombiner
+        plugin = CubeCombiner('max')
+        result = plugin.combine(req_cube_gust, req_cube_ws, 'max')
```

Comment:
I like this implementation. The only issue with which I have a concern is the lack of checking that the data are contemporaneous. The merge_cubes fails because they are not exactly the same time when using a max gust in hour cube. However I think we want to check that if the gust cube has bounds (e.g. it's a max gust cube) that the wind speed cube time falls within those bounds. If it doesn't have bounds we should still enforce time equivalence.

We could use CubeCombiner's expanded_coord keyword option to make sure the new cube has bounds that cover the period, but that get's confusing if we end up with a max in hour gust multiplied with an offset time. So I think we need a check at this level; if we don't do this then a screw up in some file passing in the suite may give something that looks like a wind gust diagnostic for the time period defined by the wind_gust_max cube (as that goes into CubeCombiner as cube1), but includes speeds from a wind field a day in the past.